proto/regen/ecocredit/v1/query.proto

```diff
@@ -73,7 +73,13 @@ service Query {
   rpc BatchesByClass(QueryBatchesByClassRequest)
       returns (QueryBatchesByClassResponse) {
     option (google.api.http).get =
-        "/regen/ecocredit/v1/classes/{class_id}/batches";
+        "/regen/ecocredit/v1/batches/class/{class_id}";
+  }
+
+  // BatchesByProject queries for all batches from a given project with pagination.
+  rpc BatchesByProject(QueryBatchesByProjectRequest) returns (QueryBatchesByProjectResponse) {
+    option (google.api.http).get =
+        "/regen/ecocredit/v1/batches/project/{project_id}";
```

Comment:
Instead of adding another query for batches by projects, we can batches query and make project_id optional.

Reply:
If we do this, I would prefer that we do this for all queries so that we are consistent.

- `ClassesByAdmin` consolidated into `Classes` with `admin` optional
- `ProjectsByReferenceId` consolidated into `Projects` with `reference_id` optional
- `BatchesByIssuer` consolidated into `Batches` with `issuer` optional
- `BatchesByClass` consolidated into `Batches` with `class_id` optional
- `BatchesByProject` consolidated into `Batches` with `project_id` optional
- `SellOrdersByBatchDenom` consolidated into `SellOrders` with `batch_denom` optional
- `SellOrdersByAddress` consolidated into `SellOrders` with `address` optional (also should be `Seller` / `seller`)

Also see https://github.com/regen-network/regen-ledger/issues/909.